### ISSUE TITLE
Should the ykllvm checkout default to `depth=1`?

### ISSUE THREAD

**Issue by ltratt** [ykjit/yk#683]
With our new setup, we're aiming to cater to users who want to build yk, but who probably don't care a great deal about ykllvm as such. However, ykllvm is *huge*. On b13 a full `git clone` took just over 5 minutes and consumes 4.2GiB. However, `clone --depth 1` takes just over 1 minute and consumes 1.9GiB.

I believe we can configure ykllvm so that, by default, it only clones to a depth of 1 with `git config -f .gitmodules submodule.ykllvm.shallow true`. This seems to me like a good idea for most users -- power users who need ykllvm's full history can re-clone with a full depth if they want without problem.

Thoughts?

**Comment by vext01** [comment]
What happens if you re-clone over a place where a sub-module is expected to be? Will it cause issues?

**Comment by vext01** [comment]
I mean, if a power uses clones their own ykllvm into a `ykllvm` dir does it confuse git?

If so, if a power user clones to `myykllvm` will a typical git workflow ever force them to clone a second ykllvm into the submodule path?

**Comment by ltratt** [comment]
I don't understand the question: why would you clone into ykllvm? Do you mean "I actually want a full clone"? You can do `git fetch --unshallow` for that.

**Comment by vext01** [comment]
I'm OK with that.

(We need to document all of this workflow)

**Comment by ltratt** [comment]
@jacob-hughes @ptersilie OK too?

**Comment by ptersilie** [comment]
Yes please! I rarely have any need to checkout earlier versions.

**Comment by jacob-hughes** [comment]
This sounds sensible to me. I do the same in Alloy and have not experienced any issues with it.
